src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/provisioning/EnterNetworkFragment.kt

```diff
@@ -121,26 +123,79 @@ class EnterNetworkFragment : Fragment() {
     try {
       ChipClient.getDeviceController().enableThreadNetwork(
           DeviceIdUtil.getLastDeviceId(requireContext()),
-          channelStr.toString().toInt(),
-          panIdStr.toString().toInt(16),
-          xpanIdStr.hexToByteArray(),
-          masterKeyStr.hexToByteArray())
+          MakeThreadOperationalDataset(
+            channelStr.toString().toInt(),
+            panIdStr.toString().toInt(16),
+            xpanIdStr.hexToByteArray(),
+            masterKeyStr.hexToByteArray()
+          ))
     } catch (e: ChipDeviceControllerException) {
       Toast.makeText(requireContext(), R.string.rendezvous_over_ble_commissioning_failure_text, Toast.LENGTH_SHORT).show()
       FragmentUtil.getHost(this, DeviceProvisioningFragment.Callback::class.java)
           ?.onCommissioningComplete(e.errorCode)
     }
   }
 
+  private fun MakeThreadOperationalDataset(
+      channel: Int,
+      panId: Int,
+      xpanId: ByteArray,
+      masterKey: ByteArray): ByteArray {
+    // Active Timestamp
+    var dataset = byteArrayOf(TYPE_ACTIVE_TIMESTAMP.toByte(), NUM_ACTIVE_TIMESTAMP_BYTES.toByte())
```

Comment:
Did you test that with the change the device can join a network created by OTBR posix? I think in the past we observed that sending Active Timestamp later than Active Timestamp of the leader causes that Mesh-Local Prefix is not retrieved from the leader, so it would be worth verifying that it's no longer the case.

Reply:
@Damian-Nordic Thanks for the consideration! I tested that the chip device can join OTBR network but I agree there will be issue when the active timestamp is later than the OTBR. I removed the Active Timestamp here.

Reply:
Thanks!